sympy/solvers/bivariate.py

```diff
@@ -165,12 +165,24 @@ def _lambert(eq, x):
     # different k values.
     lambert_real_branches = [-1, 0]
     sol = []
-    xe = Dummy('xe')
+
+    # solution of the given Lambert equation is like
+    # sol = -c/b + (a/d)*LambertW(arg, k),
+    # where arg = d/(a*b)*exp((c*d-b*f)/a/b) and k in lambert_real_branches
+    # arg can have more than one values as exp(num/p*den)[exp((c*d-b*f)/a/b)]
+    # will required to have all solutions of x**p - exp(num/den).
+
+    # calculating args for LambertW
     num, den = ((c*d-b*f)/a/b).as_numer_denom()
     p, den = den.as_coeff_Mul()
-    for e in solve(xe**p - exp(num/den), xe):
```

Comment:
or just `for e in solve(xe**p - exp(num/den), xe) if (p - 1) else [exp(num/den)]: